Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -798,6 +798,14 @@ ErrorOr<NonnullLockRefPtr<Inode>> Ext2FS::get_inode(InodeIdentifier inode) const
     return new_inode;
 }
 
+ErrorOr<void> Ext2FSInode::compute_block_list_with_exclusive_locking()
+{
+    MutexLocker block_list_locker(m_block_list_lock);
```

Comment:
Can we add a comment explaining this messy lock situation? i.e., why is it legal that some holders only lock `m_inode_lock` before accessing `m_block_list`, while others only hold `m_inode_lock`? And while we're doing that, maybe also add `VERIFY` that `m_inode_lock` is being held in a shared state, not in an exclusive state, to confirm said new comment remains true?